spec/active_force/active_query_spec.rb

```diff
@@ -132,6 +127,41 @@
     end
   end
 
+  describe '#where' do
+    before do
+      allow(client).to receive(:query).with("SELECT Id FROM table_name WHERE (Text_Label = 'foo')").and_return(api_result1)
+      allow(client).to receive(:query).with("SELECT Id FROM table_name WHERE (Text_Label = 'foo') AND (Checkbox_Label = true)").and_return(api_result2)
+    end
+    let(:api_result1) do
+      [
+        {"Id" => "0000000000AAAAABBB"},
+        {"Id" => "0000000000CCCCCDDD"},
+        {"Id" => "0000000000EEEEEFFF"}
+      ]
+    end
+    let(:api_result2) do
+      [
+        {"Id" => "0000000000EEEEEFFF"}
+      ]
+    end
+    it 'allows method chaining' do
+      result = active_query.where("Text_Label = 'foo'").where("Checkbox_Label = true")
+      expect(result).to be_a described_class
+    end
+
+    context 'when calling `where` on an ActiveQuery object that already has records' do
+      it 'returns a new ActiveQuery object' do
+        first_active_query = active_query.where("Text_Label = 'foo'")
+        first_active_query.inspect # so the query is executed
+        second_active_query = first_active_query.where("Checkbox_Label = true")
+        second_active_query.inspect
```

Comment:
why are we requiring `.inspect` to be called? i would think the expected behavior would be for this to work without requiring `.inspect`

Reply:
This is actually how `ActiveRecord::Relation` works.  The query isn't executed until the object is "used" (e.g. `to_a`, `each`, `inspect`, etc).  Otherwise you wouldn't be able to chain queries like `.where().where()`.

Reply:
i dont think this is correct